psi4/src/psi4/libfock/cubature.cc

```diff
@@ -4480,43 +4481,59 @@ void MolecularGrid::block(int max_points, int min_points, double max_radius) {
     }
 }
 
-void MolecularGrid::remove_distant_points(double Rmax) {
-    if (Rmax == std::numeric_limits<double>::max()) return;
-
-    int npoints2 = npoints_;
-    int offset = 0;
-    int point_index = 0;
-    std::vector<std::vector<MassPoint>> temp_grids(molecule_->natom());
-    for (int atom = 0; atom < atomic_grids_.size(); ++atom) {
-        for (int Q = 0; Q < atomic_grids_[atom].size(); ++Q) {
-            auto P = atomic_grids_[atom][Q];
-            Vector3 v = molecule_->xyz(atom);
-            double R = (P.x - v[0]) * (P.x - v[0]) + (P.y - v[1]) * (P.y - v[1]) + (P.z - v[2]) * (P.z - v[2]);
-            for (int A = 0; A < molecule_->natom(); A++) {
-                if (Q == A) continue;
```

Comment:
Specifically, this looks odd to me: why is the code comparing the atomic grid index with an atom index?

Reply:
here's the last edit of that block: https://github.com/psi4/psi4/commit/6141a691070c2c22aade1dd9e420679f522dfe53#diff-b6a47943fd1f813dc381d6a9e5de7b37bf671d4997497cbaa911a29a359ab66fR4473 . @hokru might be able to comment.